### PR TITLE
[Rebase & FF] 202302 Integration

### PR DESCRIPTION
**Pull request by kuqin12**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

This change integrates a series of patches checked into the overridden modules from supervisor repo.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [x] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [x] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

This was tested on QEMU virtual platform and proprietary hardware.

## Integration Instructions

End user should update to release/202302 branches for other project MU repos to consume this.

### CODE REVIEW

File: MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c [hunk 1]
```diff
@@ -2150,3 +2308,185 @@ SetUnblockRegionAttribute (
   DEBUG ((DEBUG_INFO, "%a - Exit - %r\n", __FUNCTION__, Status));
   return Status;
 }
+
+/**
+  Prevent the memory pages used for SMM page table from been overwritten.
```
Comment (kenlautner) [inline MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c:2313]: Prevent the memory pages used for SMM page table from `being` overwritten.

File: MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c [hunk 2]
```diff
@@ -68,6 +68,164 @@ EFI_MEMORY_DESCRIPTOR  *mInitMemoryMap     = NULL;
...
+    }
+  }
+
+  return TRUE;
+}
+
+/**
+  This API provides a way to allocate memory for page table.
+
+  This API can be called more once to allocate memory for page tables.
```
Comment (kenlautner) [inline MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c:181]: I assume this should be _This API can be called more_ **once** _to allocate memory for page tables._

File: MmSupervisorPkg/Drivers/MmPeiLaunchers/MmDxeSupport.inf [hunk 3]
```diff
@@ -20,7 +20,7 @@
 # Secure:
 #Track : 00000002 | MdeModulePkg/Core/PiSmmCore/PiSmmIpl.inf | 363e254ecb2e8334d7efdbe363172897 | 2022-05-13T05-08-11 | 1fe56ccdeb39aa2fb6b85a37f4fb3225f4fa2e2a
```
Comment (kenlautner) [inline MmSupervisorPkg/Drivers/MmPeiLaunchers/MmDxeSupport.inf:21]: Shouldn't this be updated as well?

File: MmSupervisorPkg/Core/MmSupervisorCore.inf [hunk 4]
```diff
@@ -11,11 +11,11 @@
 ##
 
 #Override : 00000002 | StandaloneMmPkg/Core/StandaloneMmCore.inf | afc905240e20c3c2205b0699c28e78d8 | 2023-02-14T23-02-36 | 45b2886c1f0d56d32b0b048494e4e9c2b4f88671
-#Override : 00000002 | UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuDxeSmm.inf | ba35fb6d59fa51fc56cf37ff023f3e65 | 2023-02-17T19-43-35 | 7d93781cf7750f823d93e4853e65196b132bae9f
+#Override : 00000002 | UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuDxeSmm.inf | d8c8a510e27fa589b66a6a6d6581834e | 2023-05-20T03-12-35 | dc6c1f99c459dc4107bf51d307d3484f101a95fa
 # Secure:
 #Track : 00000002 | MdeModulePkg/Core/PiSmmCore/PiSmmCore.inf | da7a5b278e02901b4917852ce56899e4 | 2023-05-03T14-04-31 | 0efbb81c016bad5b824b35713258911bb4dc5c8d
```
Comment (kenlautner) [inline MmSupervisorPkg/Core/MmSupervisorCore.inf:16]: Shouldn't this be updated as well?

File: MmSupervisorPkg/Drivers/MmPeiLaunchers/MmIplPei.inf [hunk 5]
```diff
@@ -20,7 +20,7 @@
 # Secure:
 #Track : 00000002 | MdeModulePkg/Core/PiSmmCore/PiSmmIpl.inf | 363e254ecb2e8334d7efdbe363172897 | 2022-09-02T01-34-33 | 078e685aa7aa900ea15cdf41ffe4e42bc97474d2
```
Comment (kenlautner) [inline MmSupervisorPkg/Drivers/MmPeiLaunchers/MmIplPei.inf:21]: Same as other infs.

File: MmSupervisorPkg/Drivers/MmSupervisorRing3Broker/MmSupervisorRing3Broker.inf [hunk 6]
```diff
@@ -13,7 +13,7 @@
 # Secure:
 #Track : 00000002 | MdeModulePkg/Core/PiSmmCore/PiSmmCore.inf | da7a5b278e02901b4917852ce56899e4 | 2023-05-03T14-04-31 | 0efbb81c016bad5b824b35713258911bb4dc5c8d
```
Comment (kenlautner) [inline MmSupervisorPkg/Drivers/MmSupervisorRing3Broker/MmSupervisorRing3Broker.inf:14]: Same as other infs.

File: MmSupervisorPkg/Drivers/StandaloneMmIpl/PiSmmIpl.inf [hunk 7]
```diff
@@ -21,7 +21,7 @@
 # Secure:
 #Track : 00000002 | MdeModulePkg/Core/PiSmmCore/PiSmmIpl.inf | 363e254ecb2e8334d7efdbe363172897 | 2022-05-13T05-08-11 | 1fe56ccdeb39aa2fb6b85a37f4fb3225f4fa2e2a
```
Comment (kenlautner) [inline MmSupervisorPkg/Drivers/StandaloneMmIpl/PiSmmIpl.inf:22]: Same as other infs.

**Review by kenlautner — APPROVED** [review]

File: MmSupervisorPkg/Drivers/StandaloneMmIpl/PiSmmIpl.inf (hunk 7 above)
Comment (kuqin12) [reply MmSupervisorPkg/Drivers/StandaloneMmIpl/PiSmmIpl.inf]: They are gone and changed to override tags for 202302.

File: MmSupervisorPkg/Drivers/MmSupervisorRing3Broker/MmSupervisorRing3Broker.inf (hunk 6 above)
Comment (kuqin12) [reply MmSupervisorPkg/Drivers/MmSupervisorRing3Broker/MmSupervisorRing3Broker.inf]: They are gone and changed to override tags for 202302.

File: MmSupervisorPkg/Drivers/MmPeiLaunchers/MmIplPei.inf (hunk 5 above)
Comment (kuqin12) [reply MmSupervisorPkg/Drivers/MmPeiLaunchers/MmIplPei.inf]: They are gone and changed to override tags for 202302.

File: MmSupervisorPkg/Core/MmSupervisorCore.inf (hunk 4 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/MmSupervisorCore.inf]: They are gone and changed to override tags for 202302.

File: MmSupervisorPkg/Drivers/MmPeiLaunchers/MmDxeSupport.inf (hunk 3 above)
Comment (kuqin12) [reply MmSupervisorPkg/Drivers/MmPeiLaunchers/MmDxeSupport.inf]: They are gone and changed to override tags for 202302. Thanks for catching all these.

File: MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c (hunk 1 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c]: Thanks, updated.

File: MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c (hunk 2 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c]: Yep, thanks, updated.

**Review by makubacki — APPROVED** [review]

**Pull request merged**